src/rust/channel/listener.rs

```diff
@@ -1006,7 +1006,7 @@ impl Leaving {
 }
 
 /// Errors when calling [Channel::join].
-#[derive(Clone, Debug, PartialEq, Eq, thiserror::Error)]
```

Comment:
Interesting. Was there a clippy suggestion to remove `PartialEq` and `Eq`?

Reply:
The `Arc<socket::ShutdownError>` doesn't implement those because `socket::ShutdownError` doesn't and removing it didn't break any code.  Since this is in the `rust` namespace all the callers are internal and so I know that removing `PartialEq` and `Eq` isn't a breaking change for the external API.

Reply:
I should probably be using `clippy` though across the Rust projects though.

Reply:
> The `Arc<socket::ShutdownError>` doesn't implement those because `socket::ShutdownError` doesn't and removing it didn't break any code. Since this is in the `rust` namespace all the callers are internal and so I know that removing `PartialEq` and `Eq` isn't a breaking change for the external API.

Ah this makes more sense.

Reply:
> I should probably be using `clippy` though across the Rust projects though.

I'm up for this!